src/services/refact/prompts.ts

```diff
@@ -27,7 +28,22 @@ export const promptsApi = createApi({
           credentials: "same-origin",
           redirect: "follow",
         });
-        if (result.error) return { error: result.error };
+        if (result.error) {
+          if (!isDetailMessage(result.error.data)) {
+            return {
+              error: result.error,
+            };
+          }
+          // getting first 2 lines of error message to show to user
+          const errorMessage = result.error.data.detail
+            .split("\n")
+            .slice(0, 2)
+            .join("\n");
+          api.dispatch(setError(errorMessage));
```

Comment:
Doesn't `setError` also get called in `src/app/middleware.ts`

Reply:
it does, the returned value can be found in the rejection middleware under ` action.payload.data.detail`